### DISCUSSION TITLE
How to detect that the user has closed connection?

### DISCUSSION THREAD

**Discussion by phusinte** [symfony/symfony#54935] [Q&A]
I am building a literature search website with the latest stable release of Symfony. 
Sometimes, if the user selects a lot of keywords and research filters, the server may process the request for quite a while. The issue is, if the user cancels their request (by clicking on the cross icon on their web browser), the server will keep processing their request. Which means, if they send another request, it'll be queued and will have to wait for the previous one to be finished.
That's a bit of a problem and there seem to be a few workarounds in vanilla PHP. However, I haven't seen this issue raised anywhere on Symfony.

So is there any way for me to actually detect (in a loop) that the user has closed the connection with the server, and then abort the request?

**Comment by Trismegiste** [comment]
Since you cannot be 100% sure on server-side, you'll need help from client-side. 

Look into the javascript event `window.onbeforeunload` and ask if the user want to gracefully end search queries.
This is the event typically used when editing a big form and for asking if the client is sure to lose the unsaved data.

https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event
